src/uwtools/cli.py

```diff
@@ -0,0 +1,605 @@
+    """
+    parser = add_subparser(subparsers, "run", "Run a forecast")
+    required = parser.add_argument_group(TITLE_REQ_ARG)
+    add_arg_config_file(required)
+    add_arg_model(required, choices=["FV3"])
+    optional = basic_setup(parser)
+    add_arg_dry_run(optional)
+    add_arg_quiet(optional)
+    add_arg_verbose(optional)
+
+
+def dispatch_forecast(args: Namespace) -> bool:
+    """
+    Dispatch logic for forecast mode.
+
+    :param args: Parsed command-line args.
+    """
+    forecast_class = uwtools.drivers.forecast.CLASSES[args.forecast_model]
+    forecast_class(config_file=args.config_file).run()
+    return True
+
+
+def dispatch_forecast_run(args: Namespace) -> bool:
+    """
+    Dispatch logic for forecast run submode.
+
+    :param args: Parsed command-line args.
+    """
+    raise NotImplementedError
+
+
+# Mode template
+
+
+def add_subparser_template(subparsers: Subparsers) -> None:
+    """
+    Subparser for mode: template
+
+    :param subparsers: Parent parser's subparsers, to add this subparser to.
+    """
+    parser = add_subparser(subparsers, "template", "Handle templates")
+    basic_setup(parser)
+    subparsers = add_subparsers(parser, "submode")
+    add_subparser_template_render(subparsers)
+
+
+def add_subparser_template_render(subparsers: Subparsers) -> None:
+    """
+    Subparser for mode: template render
+
+    :param subparsers: Parent parser's subparsers, to add this subparser to.
+    """
+    parser = add_subparser(subparsers, "render", "Render a template")
+    optional = basic_setup(parser)
+    add_arg_input_file(optional)
+    add_arg_output_file(optional)
+    add_arg_values_file(optional)
+    add_arg_values_needed(optional)
+    add_arg_dry_run(optional)
+    add_arg_quiet(optional)
+    add_arg_verbose(optional)
+    add_arg_key_eq_val_pairs(optional)
+
+
+def dispatch_template(args: Namespace) -> bool:
+    """
+    Dispatch logic for template mode.
+
+    :param args: Parsed command-line args.
+    """
+    return {"render": dispatch_template_render}[args.submode](args)
+
+
+def dispatch_template_render(args: Namespace) -> bool:
+    """
+    Dispatch logic for template render submode.
+
+    :param args: Parsed command-line args.
+    """
+    logging.debug("Command: %s %s", Path(sys.argv[0]).name, " ".join(sys.argv[1:]))
+    return uwtools.config.templater.render(
+        args.input_file,
+        args.output_file,
+        args.values_file,
+        dict_from_key_eq_val_strings(args.key_eq_val_pairs),
+        args.values_needed,
+        args.dry_run,
+    )
+
+
+# Arguments
+
+# pylint: disable=missing-function-docstring
+
+
+def add_arg_compare(group: Group) -> None:
+    group.add_argument(
+        "--compare",
+        action="store_true",
+        help="Compare two configs",
+    )
+
+
+def add_arg_config_file(group: Group) -> None:
+    group.add_argument(
+        "--config-file",
+        "-c",
+        help="Path to YAML config file",
```

Comment:
Yes, thank you. I'm removing the `YAML` bit from this message.

However, I've also tried in this PR to differentiate between _configs_, i.e. the files consumed by components, that guide their behavior; from simply _values_, which might be used to render templates (templates that might be configs after they're rendered) or to update the contents of non-template config files. For example, we might have a YAML file like
```
forecast:
    length: 88
```
that's used to update the values in a Fortran namelist file. This YAML file is -- if you accept my terminology -- not a _config_ of anything, since there's no component program that can consume it. It doesn't configure any component. It's just a "values file" that supplies data to `uwtools` to do things. I guess someone could say "It configures `uwtools`" but I think differentiating between components + their configs, and `uwtools` + its values would be helpful in keeping things straight. I think it also supports the idea of the shell environment or `key=value` strings on the command line being "values" sources, while it seems weird to call those two things "configs".

This difference already appears in the CLI, and I've propagated the more specific terminology into the code where it was obvious, though there may be more to do.

So, for now, I think the only places "config" appears in CLI arguments are in `uw config compare` and in `uw config validate`. The latter currently supports only YAML, but we may want to support other formats later, so being general in the help text is correct, I think. But now I'm thinking that one might want to compare or validate files that are not configs for components at all -- maybe we'd like to compare two YAML files providing values to the Jinja2 template renderer; and of course we'd want to validate the DSL file for `uwtools`. Maybe we should be less specific in the CLI about what kind of files we're dealing with, where the `uw` modes support dual use?

I almost doubted the utility of the config vs values distinction, but I think I like the fact that `uw config realize` takes arguments
```
  --input-file PATH, -i PATH
        Path to input file (defaults to stdin)
  --output-file PATH, -o PATH
        Path to output file (defaults to stdout)
  --values-file PATH
        Path to file providing override or interpolation values
```
and doesn't call the values file a "config file".

Thoughts?

Reply:
I think see what you're getting at here WRT values vs config. So we'd have users providing values...and the configuration or config files would be the result of having values applied to some input file?